Mathlib/Data/Fin/Basic.lean

```diff
@@ -2001,6 +2001,43 @@ theorem predAbove_succAbove (p : Fin n) (i : Fin n) :
     contradiction
 #align fin.pred_above_succ_above Fin.predAbove_succAbove
 
+@[simp]
+theorem succ_succAbove_predAbove {n:ℕ} ( i : Fin (n+1))  (x: Fin (n+2)) (hx : x ≠ Fin.succ i) :
+    i.succ.succAbove (i.predAbove x) = x:= by
```

Comment:
@linesthatinterlace To do what I want in this pull-request I need two theorems related to succAbove and predAbove, which I've currently put in this PR.

Do you know of an existing PR which includes these theorems?

Reply:
I have something very similar to this in my large PR, and in the refined version I'm working on. I'm fairly certain this can be done in more generality too.

Reply:
Unfortunately it's a bit of a mess. As I say I have a better version prepared with most of the lemmas but I was waiting until some of the other bits of the refactor went through to add it. I'm away house hunting for the next week but I will try to get what I have pushed so that you can look at it.

Reply:
OK, I have now pushed the improved version, #10283. This contains most of the results in #9145 (I think the main one it is missing is an antitone result on `succAbove` and some related lemmas.

I am not sure the simplex category stuff will build with this as it is currently, but it should make it all a lot easier/more direct. What I would dearly like to know is if there are any lemmas you feel are missing or improvements I can make the simplicial proofs (near the bottom of the new file) as things stand.

Reply:
@linesthatinterlace Thank you for this, I will merge #10283 with this PR and see how I can utilise the results. I hope your house hunting goes well!